airflow/utils/dates.py

```diff
@@ -257,7 +257,9 @@ def days_ago(n, hour=0, minute=0, second=0, microsecond=0):
     Get a datetime object representing `n` days ago. By default the time is
     set to midnight.
     """
-    today = timezone.utcnow().replace(hour=hour, minute=minute, second=second, microsecond=microsecond)
+    today = datetime.now(timezone.TIMEZONE).replace(
+        hour=hour, minute=minute, second=second, microsecond=microsecond
+    )
```

Comment:
Maybe

```python
datetime.combine(
    date.today(),
    time(hour, minute, second, microsecond, tzinfo=timezone.TIMEZONE),
)
```

is easier to understand?

Reply:
Or even

```python
def days_ago(n, hour=0, minute=0, second=0, microsecond=0):
    return datetime.combine(
        date.today() - timedelta(days=n),
        time(hour, minute, second, microsecond, tzinfo=timezone.TIMEZONE),
    )
```

Reply:
`date.today()` is UTC today(). So e.g. when it is `2021-01-12 00:00` in UTC, date.today() returns `2021-01-12` even though it is still `2021-01-11` in Los Angeles time. `today()` is classmethod and does not take any other argument, so we can just use `datetime.now()` for date parameter in `datetime.combine()`.

```python
def days_ago(n, hour=0, minute=0, second=0, microsecond=0):
    return datetime.combine(
        datetime.now(timezone.TIMEZONE) - timedelta(days=n),
        time(hour, minute, second, microsecond, tzinfo=timezone.TIMEZONE),
    )
```

Reply:
> date.today() returns today in UTC.

It’s documented to return the local date, so if it really returns the UTC time’s date as you describe, there’s a serious bug in Python. But all those ultimately depend on `time.time()` anyway, so let’s use your version since you already made sure it works 🙂

Reply:
> It’s documented to return the local date  

My wrong, I was confused about this. `date.today()` does returns the system local date, but i think we still needs to respect timezone settings in airflow.cfg, which is `timezone.TIMEZONE`, not system local date.